contracts/potlock/src/potlock_admin_interactions.rs

```diff
@@ -0,0 +1,132 @@
+{
+    #[only_admin]
+    #[endpoint(changeFeeForPots)]
+    fn change_fee_for_pots(&self, token_identifier: TokenIdentifier, fee: BigUint) {
+        require!(
+            token_identifier.is_valid_esdt_identifier(),
+            "Invalid token provided"
+        );
+        require!(fee > 0, "Amount is 0");
+        self.fee_token_identifier().set_if_empty(&token_identifier);
+        self.fee_amount().set_if_empty(fee);
+    }
+
+    #[only_admin]
+    #[endpoint(acceptPot)]
+    fn accept_pot(&self, potlock_id: PotlockId) {
+        self.require_potlock_exists(potlock_id);
+        self.require_potlock_is_inactive(potlock_id);
+
+        let mut accepted_potlock = self.potlocks().get(potlock_id);
+        accepted_potlock.status = Status::Active;
+        self.potlocks().set(potlock_id, &accepted_potlock);
+    }
+
+    #[only_admin]
+    #[endpoint(removePot)]
+    fn remove_pot(&self, potlock_id: PotlockId) {
+        self.require_potlock_exists(potlock_id);
+        self.require_potlock_is_inactive(potlock_id);
+
+        let potlock_mapper = self.potlocks();
+        let pot_proposer = potlock_mapper.get(potlock_id).proposer;
+        let fee_pot_payment = EsdtTokenPayment::new(
+            self.fee_token_identifier().get(),
+            0u64,
+            self.fee_amount().get(),
+        );
+
+        self.send()
+            .direct_non_zero_esdt_payment(&pot_proposer, &fee_pot_payment);
+        self.potlocks().clear_entry(potlock_id);
+    }
+
+    #[only_admin]
+    #[endpoint(acceptApplication)]
+    fn accept_application(&self, project_id: ProjectId) {
+        self.require_project_exists(project_id);
+        self.require_project_is_inactive(project_id);
+
+        let mut accepted_project = self.projects().get(project_id);
+        accepted_project.status = Status::Active;
+        self.projects().set(project_id, &accepted_project);
+    }
+
+    #[only_admin]
+    #[endpoint(removeApplication)]
+    fn remove_application(&self, project_id: ProjectId) {
+        self.require_project_exists(project_id);
+        self.require_project_is_active(project_id);
+
+        let mut rejected_project = self.projects().get(project_id);
+        rejected_project.status = Status::Inactive;
+        self.projects().set(project_id, &rejected_project);
+    }
+
+    #[only_admin]
+    #[endpoint(rejectDonation)]
+    fn reject_donation(&self, potlock_id: PotlockId, user: ManagedAddress) {
+        self.require_potlock_exists(potlock_id);
+        let opt_fee_pot_payment = self.pot_donations(potlock_id).get(&user);
+
+        require!(opt_fee_pot_payment.is_some(), "No donation for this user");
+        let fee_pot_payment = unsafe { opt_fee_pot_payment.unwrap_unchecked() };
+
+        self.send()
+            .direct_non_zero_esdt_payment(&user, &fee_pot_payment);
+        self.pot_donations(potlock_id).remove(&user);
+    }
+
+    #[only_admin]
+    #[endpoint(distributePotToProjects)]
+    fn distribute_pot_to_projects(
```

Comment:
A few more questions here. Not mandatory but are good to be discussed.
- Right now, there is no way to be sure all the projects that applied for this pot were covered in the provided array, is it ok for the owner to be able to give the amounts arbitrary and to not all projects? 
- Also, there is no check that the project applied to this specific pot, maybe add a check after you read each project from storage, that the project is active for this pot? If not, why even bother to approve a project for a pot?
- And finally, how about a limit to the number of projects that can apply for a pot? Either here for the number of elements for project_percentages, or for when subscribing a project to a pot.